### DISCUSSION TITLE
How can I add a load path before running a script?

### DISCUSSION THREAD

**Discussion by merulus** [nushell/nushell#12836] [Q&A]
The `NU_LIB_DIRS` set in the environment doesn't seem to be read?

I tried:
```bash
export NU_LIB_DIRS=/path/to/lib
nu script.nu
```

but it cannot find the modules under /path/to/lib

**Comment by fdncred** [comment]
iirc, `nu -l script.nu` will load both the env.nu and the config.nu.

**Reply by merulus** [reply]
This requires a separate config file, which is inconvenient.
And I need to dynamically add a path that is unknown until runtime.

**Comment by ghost** [comment]
if you look at `nu --help` you can find the following argument:
```nu
  -I, --include-path <String> - set the NU_LIB_DIRS for the given script (delimited by char record_sep (''))
```

**Reply by merulus** [reply]
This only works with `-c, --commands`, not scripts. Or did I miss something?

**Reply by ghost** [reply]
in nu 0.92.1 it just says `script`, which i assumed to include a script attached as argument.

but after testing it i agree that it dosn't work. it actually dosn't work at all for me:
```
$ "print $env.NU_LIB_DIRS" | save -r b.nu

$ nu b.nu
╭───┬────────────────────────────────────╮
│ 0 │ /home/user/.config/nushell/scripts │
╰───┴────────────────────────────────────╯

$ nu -I /dev/null b.nu
╭───┬────────────────────────────────────╮
│ 0 │ /home/user/.config/nushell/scripts │
╰───┴────────────────────────────────────╯

$ nu -I /dev/null -c 'print $env.NU_LIB_DIRS'
╭───┬────────────────────────────────────╮
│ 0 │ /home/user/.config/nushell/scripts │
╰───┴────────────────────────────────────╯

$ nu -I /dev/null -c (open -r b.nu)
╭───┬────────────────────────────────────╮
│ 0 │ /home/user/.config/nushell/scripts │
╰───┴────────────────────────────────────╯
```
